docs/TPM2_PCR_MEASUREMENTS.md

```diff
@@ -66,7 +66,7 @@ below for details on PE section measurements done by `systemd-stub`.)
 UTF-16.
 
 → **Measured hash** covers the literal kernel command line in UTF-16 (without any
-trailing NUL bytes).
+trailing U+0000 codepoints).
```

Comment:
I think the previous text was better. "No NUL bytes" means that all the zeros are cut off. But "no U+0000 codepoints" superficially means that if the end was e.g. `NUL NUL NUL`, then only `NUL NUL` would be cut off, and we would measure the the remaining unpaired `NUL`. This is all a bit theoretical, since somebody would need to give as a buffer with an odd number of bytes, but meh, I think it's nicer to cover this case and say that all NUL bytes will be removed.

Reply:
well, the problem that arises is that this sounds so far if you have a codepoint with the upper bits set to 0, then we are dropping them, which we of course do not.

Reply:
i mean i am open to change the wording to something else, but I am pretty sure the current wording is pretty bad, and at least @grawity seems to agree, as he brought it up ;-)

Reply:
> I think the previous text was better. "No NUL bytes" means that all the zeros are cut off. But "no U+0000 codepoints" superficially means that if the end was e.g. `NUL NUL NUL`, then only `NUL NUL` would be cut off, and we would measure the the remaining unpaired `NUL`. This is all a bit theoretical, since somebody would need to give as a buffer with an odd number of bytes,

It was prompted by a practical issue in tpm_futurepcr, where I wanted to "replay" my own events that would match ones that systemd-boot would generate, and had difficulties until I realized I had to deliberately chop off the last NUL and measure a UTF-16 buffer with an odd number of bytes myself, as it turned out that was what systemd-boot had been doing. (And then in some release the behavior changed to correctly keeping a full codepoint... and I *think* then later it changed again to truncating.)